rules/common/src/main/kotlin/io/nlopez/compose/rules/ViewModelForwarding.kt

```diff
@@ -58,27 +63,82 @@ class ViewModelForwarding : ComposeKtVisitor {
+                    ) {
+                        // Get VALUE_ARGUMENT that has a REFERENCE_EXPRESSION. This would map to `viewModel` in this example:
+                        // MyComposable(viewModel, ...)
+                        callExpression.valueArguments
+                            .mapNotNull { valueArgument ->
+                                when (val argumentExpression = valueArgument.getArgumentExpression()) {
+                                    is KtReferenceExpression, is KtThisExpression -> argumentExpression
+                                    else -> null
+                                }
+                            }
+                            .filter { reference ->
+                                reference.text in viewModelParameterNames ||
+                                    (
+                                        reference.text == "it" &&
+                                            scopedParameter in viewModelParameterNames && usesItObjectRef
+                                        ) ||
+                                    (
+                                        reference.text == "this" &&
+                                            scopedParameter in viewModelParameterNames && !usesItObjectRef
+                                        )
+                            }
+                            .forEach { _ ->
+                                emitter.report(callExpression, AvoidViewModelForwarding, false)
+                            }
+                    }
+
+                    val scopedFunctions = setOf("with", "apply", "run", "also", "let")
+                    // Check if the call is a scope function
+                    if (callExpression.calleeExpression is KtNameReferenceExpression &&
+                        (callExpression.calleeExpression as KtNameReferenceExpression)
+                            .getReferencedName() in scopedFunctions
+                    ) {
+                        callExpression.lambdaArguments
+                            .mapNotNull { it.getLambdaExpression()?.bodyExpression }
+                            .forEach { lambdaBodyExpression ->
+                                checkCallExpressions(
+                                    bodyExpression = lambdaBodyExpression,
+                                    scopedParameter = callExpression.getScopedParameterValue(scopedParameter),
+                                    usesItObjectRef = callExpression.hasItObjectReference,
+                                )
+                            }
+                    }
+                }
+        }
+        checkCallExpressions(bodyBlock)
+    }
+
+    private val KtCallExpression.isWithScope: Boolean
+        get() = (calleeExpression as? KtNameReferenceExpression)?.getReferencedName() == "with"
+
+    private val KtCallExpression.hasItObjectReference: Boolean
+        get() = (calleeExpression as? KtNameReferenceExpression)?.getReferencedName() in setOf("let", "also")
```

Comment:
This set should be in a companion object, so that it doesn't get instantiated everytime this method is invoked.